python/tskit/trees.py

```diff
@@ -5098,13 +5098,27 @@ def mutation(self, id_):
             metadata_decoder=self.table_metadata_schemas.mutation.decode_row,
         )
 
-    def site(self, id_):
+    def site(self, id_=None, position=None):
         """
         Returns the :ref:`site <sec_site_table_definition>` in this tree sequence
-        with the specified ID.
+        with either the specified ID or position.
```

Comment:
An explanation of the scaling characteristics would be good here as the caller should be informed that this method when called with `position=` does a binary search of the position array. We don't want people to iterate through positions to get sites!